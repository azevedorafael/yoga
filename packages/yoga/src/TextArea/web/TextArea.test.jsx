import React from 'react';
import { render } from '@testing-library/react';

import { ThemeProvider, TextArea } from '../..';

describe('<TextArea />', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <ThemeProvider>
        <TextArea />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
