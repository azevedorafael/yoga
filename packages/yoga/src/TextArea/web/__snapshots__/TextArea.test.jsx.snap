// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextArea /> should match snapshot 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 4px;
  color: #9898A7;
}

.c0 {
  display: inline-block;
  cursor: text;
  width: auto;
}

.c1 {
  position: relative;
  display: inline-block;
  vertical-align: top;
  height: 52px;
  border-radius: 8px;
  border-width: 1px;
  border-style: solid;
  border-color: #D3D3E2;
  height: 88px;
  padding-top: 16px;
  padding-right: 16px;
  padding-bottom: 16px;
  padding-left: 16px;
}

.c1 {
  width: 320px;
}

.c1 svg {
  position: absolute;
  top: 0;
  right: 0;
  padding-right: 16px;
  padding-left: 8px;
  height: 52px;
  width: 20px;
  fill: #9898A7;
  outline: none;
  box-sizing: content-box;
  cursor: pointer;
}

.c1 svg:hover,
.c1 svg:focus {
  fill: #41414A;
}

.c1:focus-within legend {
  max-width: 1000px;
  font-weight: 700;
}

.c1:hover,
.c1:focus-within {
  border-color: #41414A;
}

.c2 {
  width: 100%;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: transparent;
  outline: none;
  height: 100%;
  padding-top: 16px;
  padding-right: 16px;
  padding-bottom: 16px;
  padding-left: 16px;
  color: #41414A;
  border: none;
  font-family: Open Sans,sans-serif;
  font-size: 14px;
  font-weight: 400;
  box-sizing: border-box;
  resize: none;
  height: 100%;
  padding: 0;
}

.c2:focus {
  color: #41414A;
}

.c2:focus + label {
  top: 0;
  left: 16px;
  font-size: 12px;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  -webkit-transition-duration: 500ms;
  transition-duration: 500ms;
  -webkit-transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  transition-timing-function: cubic-bezier(0,0.75,0.1,1);
  font-weight: 700;
  color: #41414A;
}

.c2:disabled {
  cursor: not-allowed;
  color: #D3D3E2;
}

.c2::-webkit-input-placeholder {
  color: #9898A7;
}

.c2::-moz-placeholder {
  color: #9898A7;
}

.c2:-ms-input-placeholder {
  color: #9898A7;
}

.c2::placeholder {
  color: #9898A7;
}

.c2[type="number"]::-webkit-outer-spin-button,
.c2[type="number"]::-webkit-inner-spin-button {
  display: none;
}

.c2:invalid {
  box-shadow: none;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      label=""
    >
      <textarea
        class="c2"
        label=""
      />
      
    </div>
    <div
      class="c3"
    />
  </div>
</div>
`;
